lib/src/worker/worker.dart

```diff
@@ -0,0 +1,117 @@
+// ignore_for_file: public_member_api_docs
+
+import 'dart:async';
+import 'dart:convert' show jsonEncode;
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+// ignore: avoid_web_libraries_in_flutter
+import 'dart:js_util';
```

Comment:
Why this ignore? Can you maybe add a comment to the file that explains why we need this and how we can potentially, in the future, get rid of this?

Reply:
`jsSendMessage()`, `callbackToStream()` and `class Worker` are only used by `web/worker.dart`. Beeing it stored in the `lib/worker` dir, the linter thinks this source is used for deployment on other platforms too, but it's not.
It's worth to try moving these in the `web/worker.dart` code. I will try before merging.

Anyway, we should also rake a note that `dart:html` will be [deprecated in the future](https://dart.dev/interop/js-interop). We use it in the `Loader` for example. Likewise the `dart:js_util` used here but I didn't find a way to use js_interop instead.

Reply:
Got it.

If anyone with `js_interop` experience is reading this PR, then — first of all, hi! Thanks for having a look! If you can, please do review this file in particular.

Reply:
I moved from here what was needed only in `web/worker.dart`. The `class Worker`, `callbackToStream()` and `jsSendMessage()` are now in `web/worker.dart`.

Also the `dart:js_util` has been replaced by `dart:js_interop_unsafe` and the code adjusted to support it. The ignore is now gone.